### ISSUE TITLE
CPU 100% after update to v0.20.1

### ISSUE THREAD

**Issue by kotevskimk** [AzuraCast/AzuraCast#7207]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

v0.20.1 Stable 

### What happened?

I updated Azuracast to v0.20.1 Stable and CPU now is 100%.
Nothing changed before, everything was working fine, and CPU was 20-30%.
I have VPS with 8 CPU and 30GB of ram.
I have previus Stable version, and worked without any errors.

### Relevant log output

```shell
[2024-06-17T00:24:13.511699+00:00] AzuraCast.NOTICE: Starting new Message Queue worker process. {"runtime":0,"workerName":"app_worker_0"} []
[2024-06-17T00:24:18.629140+00:00] AzuraCast.ERROR: Message queue error: NOGROUP No such key 'messages.high_priority' or consumer group 'symfony' in XREADGROUP with GROUP option {"workerName":"app_worker_0","exception":"[object] (Symfony\\Component\\Messenger\\Exception\\TransportException(code: 0): NOGROUP No such key 'messages.high_priority' or consumer group 'symfony' in XREADGROUP with GROUP option at /var/azuracast/www/vendor/symfony/redis-messenger/Transport/Connection.php:447)"} []
[2024-06-17T00:24:18.630109+00:00] AzuraCast.WARNING: Console command `azuracast:queue:process` exited with error code 1. [] []
[2024-06-17T00:24:19.934942+00:00] AzuraCast.NOTICE: Failed to define an expiration for the "{resource}" lock, someone else acquired the lock. {"resource":{"Symfony\\Component\\Lock\\Key":"lock_nowplaying_ragostivar"}} []
[2024-06-17T00:24:19.937851+00:00] AzuraCast.ERROR: Symfony\Component\Lock\Exception\LockConflictedException:  (uncaught exception) at /var/azuracast/www/vendor/symfony/lock/Store/RedisStore.php line 166 while running console command `azuracast:sync:nowplaying` {"exception":"[object] (Symfony\\Component\\Lock\\Exception\\LockConflictedException(code: 0):  at /var/azuracast/www/vendor/symfony/lock/Store/RedisStore.php:166)"} []
[2024-06-17T00:24:19.938180+00:00] AzuraCast.WARNING: Console command `azuracast:sync:nowplaying` exited with error code 1. [] []
[2024-06-17T00:24:21.238153+00:00] AzuraCast.NOTICE: Starting new Message Queue worker process. {"runtime":0,"workerName":"app_worker_0"} []
[2024-06-17T00:24:31.370576+00:00] AzuraCast.WARNING: Duplicate prevention yielded no playable song; resetting song queue. [] {"station":{"id":4,"name":"Radio Pela"}}
[2024-06-17T00:24:31.372124+00:00] AzuraCast.WARNING: Playlist "default" did not return a playable track. {"playlist_id":13,"playlist_order":"shuffle","allow_duplicates":false} {"station":{"id":4,"name":"Radio Pela"}}
[2024-06-17T00:24:31.972462+00:00] AzuraCast.ERROR: Request returned status code 400. {"body":""} {"station":{"id":3,"name":"Radio G"}}
```

**Comment by asa767** [comment]
Which vps provider are you using. I am facing same situation with Contabo.

**Comment by kotevskimk** [comment]
Yes Contabo.

Everything goes well untill now.
But from the last update the problem started.

About Contabo, they changed packages, now this instance goes with 24GB of ram. And I think they Do this to force me to change the package, in some way. Or I am only thonking. ;)

**Comment by BusterNeece** [comment]
We'd need to see something like `htop` to see what's actually consuming the CPU.

Did you enable any of the new features, like ReplayGain or AutoCue? Those will result in high initial CPU load. You can lower some of the CPU load by disabling "Write Playlists to Liquidsoap" on the AutoDJ tab of the station profile.

**Comment by handris** [comment]
![spike](https://github.com/AzuraCast/AzuraCast/assets/3883243/52a1fa17-6a64-4924-a992-8a2adbc34b8e)
Hi,
I have some problem on real machine with v0.20.1 Stable. So for production use I keep on v0.19.5 wich very stable and use low cpu.  By the way there is no option to disable "Write Playlists to Liquidsoap"  on v0.20.1. I hope this will fix on next stable version.

**Comment by handris** [comment]
![spike](https://github.com/AzuraCast/AzuraCast/assets/3883243/5ff294a9-fca5-4d5f-9ac1-2997c69b96ce)
Hi,
I have some problem on real machine with v0.20.1 Stable. So for production use I keep on v0.19.5 wich very stable and use low cpu.  By the way there is no option to disable "Write Playlists to Liquidsoap"  on v0.20.1. I hope this will fix on next stable version.

**Comment by asa767** [comment]
<img width="1380" alt="Ekran Resmi 2024-06-18 10 07 18" src="https://github.com/AzuraCast/AzuraCast/assets/26309435/503bf2ab-69d0-493f-83ef-446017404b57">
On the left from Contabo 8 cpu 16gb ram, on the right 4 cpu 8gb ram. It seems issue related to vps provider but on the other hand cpu usage increased after 0.20.1 update. My setup is 18 stations with total of 19 mount points.

**Comment by renatopinto89** [comment]
From my own experience, Contabo is absolute trash when it comes to shared VPS.
They oversell the nodes to oblivion resulting on LOTS of CPU steal.

With 18 stations I would fully recommend Bare Metal on Hetzner or OVH (never tried Contabos Dedicated offerings, so can´t tell if they´re good or not) in that aspect.

Best Regards

**Comment by handris** [comment]
> Reference in
Hi Tom, on version on v0.20.1  there is no option to disable : Write Playlists to Liquidsoap. Not yet available

**Comment by kotevskimk** [comment]
Is that option availabne on Roling Release. If it is, do I have to update from Stable to Rolling Release.

**Comment by handris** [comment]
> Is that option availabne on Roling Release. If it is, do I have to update from Stable to Rolling Release.

Yes.  If you dont have choice to revert back by image backup. Better upgrade to rollong release.
Rolling Release #[6c6e84e] seems better now. I tested today.

**Comment by kotevskimk** [comment]
Yes I have backups, but without media files.
If I do backup, the version will be the same Stable, last updated. 
But I thing that I have made Snapshot on Contabo.
Or I will make mess with everything?
OOhhh, what to do.

**Comment by exbiz-galban** [comment]
I experienced the same issue. Everything was working perfectly, with CPU usage between 20 to 30%. After updating to version 0.20.1, the CPU usage spiked to 100% constantly. I tried rolling back, but encountered only errors. I had to delete everything and create a new setup using version 0.20.0.

P.S. The problem is not 'CPU steal' from Contabo, as everything was running normally for months before the update.

**Comment by kotevskimk** [comment]
Just tell me how do i choose which version to install. 0.20.0  It's on
setup or?

**Comment by t0mtaylor** [comment]
@kotevskimk I found you have to delete some of the docker volumes before switching between stable and rolling release, as your station setup is stored in the database - then the volumes are recreated fresh:

See the steps here https://github.com/AzuraCast/AzuraCast/issues/7162#issuecomment-2143803893

**Comment by kotevskimk** [comment]
Dear All,
Thank You for all Your support.
I have made a Snapshot on Contabo Control panel before making upgrade. I just revert to that snapshot, an You won't believe All thing was made for 3 seconds.
Then I restarted the VPS, and MIRACLE.
AzuraCast is on earlier Stable version that worked perfectly.
Thank You!

**Comment by istahilagi** [comment]
> Write Playlists to Liquidsoap

How to do it?

**Comment by asa767** [comment]
> <img alt="Ekran Resmi 2024-06-18 10 07 18" width="1380" src="https://private-user-images.githubusercontent.com/26309435/340575807-503bf2ab-69d0-493f-83ef-446017404b57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxMjc1NjgsIm5iZiI6MTcxOTEyNzI2OCwicGF0aCI6Ii8yNjMwOTQzNS8zNDA1NzU4MDctNTAzYmYyYWItNjlkMC00OTNmLTgzZWYtNDQ2MDE3NDA0YjU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIzVDA3MjEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlkOWNiMTFmOTc3OTc1ODdiYjY4MmNhZmM1ZDc5M2Q2ODEyOTBmN2IxMDMzMWE3NzM0YmQ3MTA1OWI2ZDdjY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z8WIExBjIka_f2enxGalHPW4DciJkWOB8WU1vGPIYNk"> On the left from Contabo 6 cpu 16gb ram, on the right Digital Ocean 4 cpu 8gb ram. It seems issue related to vps provider but on the other hand cpu usage increased after 0.20.1 update. My setup is 18 stations with total of 19 mount points.

Hi, 
There are some uptdates about my Contabo setup with 18 stations. I've just receivede an email from Contabo support saying "Please be informed that we checked the situation and were able to adjust some settings. The performance of your server should improve accordingly. Kindly check from your side again." After that, taa taaa:
<img width="689" alt="Ekran Resmi 2024-06-23 10 18 24" src="https://github.com/AzuraCast/AzuraCast/assets/26309435/806970be-f917-4a67-8f4c-198cf0ca2e6f">

**Comment by ATOMONLINE** [comment]
I installed on OS Ubuntu 22.04 LTS [64 Bit] 100% no CPU issues.

![Screenshot_4](https://github.com/AzuraCast/AzuraCast/assets/68115153/5a8b7214-255d-486b-80d5-a49d2ed38a37)

**Comment by Moonbase59** [comment]
Yup, sad business practices nowadays, probably due to the overall el-cheapo mentality. Providers advertise unrealistic price/performance, tend to not keep their promises, and "adjust" for customers that complain.

In my days, we always slightly _over-achieved_, and customers were happy. And came back.

**Comment by kotevskimk** [comment]
> > <img alt="Ekran Resmi 2024-06-18 10 07 18" width="1380" src="https://private-user-images.githubusercontent.com/26309435/340575807-503bf2ab-69d0-493f-83ef-446017404b57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxMjc1NjgsIm5iZiI6MTcxOTEyNzI2OCwicGF0aCI6Ii8yNjMwOTQzNS8zNDA1NzU4MDctNTAzYmYyYWItNjlkMC00OTNmLTgzZWYtNDQ2MDE3NDA0YjU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIzVDA3MjEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlkOWNiMTFmOTc3OTc1ODdiYjY4MmNhZmM1ZDc5M2Q2ODEyOTBmN2IxMDMzMWE3NzM0YmQ3MTA1OWI2ZDdjY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z8WIExBjIka_f2enxGalHPW4DciJkWOB8WU1vGPIYNk"> On the left from Contabo 6 cpu 16gb ram, on the right Digital Ocean 4 cpu 8gb ram. It seems issue related to vps provider but on the other hand cpu usage increased after 0.20.1 update. My setup is 18 stations with total of 19 mount points.
> 
> Hi, There are some uptdates about my Contabo setup with 18 stations. I've just receivede an email from Contabo support saying "Please be informed that we checked the situation and were able to adjust some settings. The performance of your server should improve accordingly. Kindly check from your side again." After that, taa taaa: <img alt="Ekran Resmi 2024-06-23 10 18 24" width="687" src="https://private-user-images.githubusercontent.com/26309435/342043327-0c2f8712-7450-4897-ac16-7a83d8a5d045.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxMzQ5MzksIm5iZiI6MTcxOTEzNDYzOSwicGF0aCI6Ii8yNjMwOTQzNS8zNDIwNDMzMjctMGMyZjg3MTItNzQ1MC00ODk3LWFjMTYtN2E4M2Q4YTVkMDQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIzVDA5MjM1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzZjI3Yzg2ZTUwMmMwM2ZjOWRhMmM5ODlkYzQyMGNkZTkwNjVhOGZkMjY1YWMyMmYwNWRkMWE2NWU2ODUzNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OsSdkSfW4BGPayWl6DEdOiF1bBL0XP5qVjQywYyA-WQ">

You like to tell me that everything was from Contabo, the problem is Not the 0.20.1 update?
OMG!
You have Stable 0.20.1 version working great after Contabo adjust some settings?
Please tell me.

**Comment by asa767** [comment]
> > > <img alt="Ekran Resmi 2024-06-18 10 07 18" width="1380" src="https://private-user-images.githubusercontent.com/26309435/340575807-503bf2ab-69d0-493f-83ef-446017404b57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxMjc1NjgsIm5iZiI6MTcxOTEyNzI2OCwicGF0aCI6Ii8yNjMwOTQzNS8zNDA1NzU4MDctNTAzYmYyYWItNjlkMC00OTNmLTgzZWYtNDQ2MDE3NDA0YjU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIzVDA3MjEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlkOWNiMTFmOTc3OTc1ODdiYjY4MmNhZmM1ZDc5M2Q2ODEyOTBmN2IxMDMzMWE3NzM0YmQ3MTA1OWI2ZDdjY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z8WIExBjIka_f2enxGalHPW4DciJkWOB8WU1vGPIYNk"> On the left from Contabo 6 cpu 16gb ram, on the right Digital Ocean 4 cpu 8gb ram. It seems issue related to vps provider but on the other hand cpu usage increased after 0.20.1 update. My setup is 18 stations with total of 19 mount points.
> 
> > 
> 
> > Hi, There are some uptdates about my Contabo setup with 18 stations. I've just receivede an email from Contabo support saying "Please be informed that we checked the situation and were able to adjust some settings. The performance of your server should improve accordingly. Kindly check from your side again." After that, taa taaa: <img alt="Ekran Resmi 2024-06-23 10 18 24" width="687" src="https://private-user-images.githubusercontent.com/26309435/342043327-0c2f8712-7450-4897-ac16-7a83d8a5d045.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxMzQ5MzksIm5iZiI6MTcxOTEzNDYzOSwicGF0aCI6Ii8yNjMwOTQzNS8zNDIwNDMzMjctMGMyZjg3MTItNzQ1MC00ODk3LWFjMTYtN2E4M2Q4YTVkMDQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIzVDA5MjM1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzZjI3Yzg2ZTUwMmMwM2ZjOWRhMmM5ODlkYzQyMGNkZTkwNjVhOGZkMjY1YWMyMmYwNWRkMWE2NWU2ODUzNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OsSdkSfW4BGPayWl6DEdOiF1bBL0XP5qVjQywYyA-WQ">
> 
> 
> 
> You like to tell me that everything was from Contabo, the problem is Not the 0.20.1 update?
> 
> OMG!
> 
> You have Stable 0.20.1 version working great after Contabo adjust some settings?
> 
> Please tell me.

As mentioned above, Contabo throttle down and Azuracast updates came together. I updated to rolling release channel and closed all cpu killing settings but nothing changed. I thought the issue was because of Contabo. 

Also my radio stations was facing interruptions time to time. I think it's also because of Contabo. So I decided to move my stations to a dedicated server. Will see what happens.

**Comment by istahilagi** [comment]
I have also been using Contabo for more than 3 years. I feel that recently some of my VPS in Contabo have decreased in performance...

**Comment by renatopinto89** [comment]
My problem with Contabo which made me look elsewhere is the overselling.
Your VPS can be working just fine for a while until you have the bad luck of getting a "noisy neighbor" that will hog the resources in the node, making all other VMs lose those resources. This shouldn't be a problem, but Contabo tends to not properly scale and balance resources, resulting on all of the sudden, your VPS starts lagging all over the place for seemingly no reason.
It's just bad business practice in a nutshell, but at the same time, you get what you pay for

**Comment by kotevskimk** [comment]
> > > > <img alt="Ekran Resmi 2024-06-18 10 07 18" width="1380" src="https://private-user-images.githubusercontent.com/26309435/340575807-503bf2ab-69d0-493f-83ef-446017404b57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxMjc1NjgsIm5iZiI6MTcxOTEyNzI2OCwicGF0aCI6Ii8yNjMwOTQzNS8zNDA1NzU4MDctNTAzYmYyYWItNjlkMC00OTNmLTgzZWYtNDQ2MDE3NDA0YjU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIzVDA3MjEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlkOWNiMTFmOTc3OTc1ODdiYjY4MmNhZmM1ZDc5M2Q2ODEyOTBmN2IxMDMzMWE3NzM0YmQ3MTA1OWI2ZDdjY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z8WIExBjIka_f2enxGalHPW4DciJkWOB8WU1vGPIYNk"> On the left from Contabo 6 cpu 16gb ram, on the right Digital Ocean 4 cpu 8gb ram. It seems issue related to vps provider but on the other hand cpu usage increased after 0.20.1 update. My setup is 18 stations with total of 19 mount points.
> > 
> > 
> > > 
> > 
> > 
> > > Hi, There are some uptdates about my Contabo setup with 18 stations. I've just receivede an email from Contabo support saying "Please be informed that we checked the situation and were able to adjust some settings. The performance of your server should improve accordingly. Kindly check from your side again." After that, taa taaa: <img alt="Ekran Resmi 2024-06-23 10 18 24" width="687" src="https://private-user-images.githubusercontent.com/26309435/342043327-0c2f8712-7450-4897-ac16-7a83d8a5d045.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxMzQ5MzksIm5iZiI6MTcxOTEzNDYzOSwicGF0aCI6Ii8yNjMwOTQzNS8zNDIwNDMzMjctMGMyZjg3MTItNzQ1MC00ODk3LWFjMTYtN2E4M2Q4YTVkMDQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIzVDA5MjM1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzZjI3Yzg2ZTUwMmMwM2ZjOWRhMmM5ODlkYzQyMGNkZTkwNjVhOGZkMjY1YWMyMmYwNWRkMWE2NWU2ODUzNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OsSdkSfW4BGPayWl6DEdOiF1bBL0XP5qVjQywYyA-WQ">
> > 
> > 
> > You like to tell me that everything was from Contabo, the problem is Not the 0.20.1 update?
> > OMG!
> > You have Stable 0.20.1 version working great after Contabo adjust some settings?
> > Please tell me.
> 
> As mentioned above, Contabo throttle down and Azuracast updates came together. I updated to rolling release channel and closed all cpu killing settings but nothing changed. I thought the issue was because of Contabo.
> 
> Also my radio stations was facing interruptions time to time. I think it's also because of Contabo. So I decided to move my stations to a dedicated server. Will see what happens.

Did You try the dedicated server. Did Contabo was reson for AzuraCast? Or it is something else. Please write.

**Comment by asa767** [comment]
Hi [kotevskimk](https://github.com/kotevskimk),
I've just delivered an AX42 server from hetzner.com. Performance is not comparable with Contabo's vps. Now load is below 1 with same setup. It's amazing. On the other hand as I said above same setup had worked 10x better on DigitalOcean during this period. 
<img width="626" alt="Ekran Resmi 2024-06-26 19 18 37" src="https://github.com/AzuraCast/AzuraCast/assets/26309435/b80333f4-c2f6-4c02-ae41-5463f767ee7c">

**Comment by kotevskimk** [comment]
And which version You install, newone Stable 20.1 from AzuraCast od Roling Release?

**Comment by asa767** [comment]
0.20.1 Stable

**Comment by keybrdist** [comment]
Same issue on AWS t3.large, but worse. Constant 100% CPU on 0.20.1 stable.

I was able to run at about 70% CPU in some previous versions on a t3.medium. I updated to large to try to fix my issue but now it is consuming >100% CPU on the t3.large.

<img width="699" alt="image" src="https://github.com/AzuraCast/AzuraCast/assets/3534206/32414002-dffd-4f71-8325-398a4c069b3f">

Some php processes like `php bin/console azuracast:sync:run` and `sync:nowplaying` are using constant >20% .. weird. 
`ubuntu   2364427 22.7  1.8 391112 73668 ?        R    18:04   0:00 php /var/azuracast/www/bin/console azuracast:sync:nowplaying:station dnbradio
ubuntu   2364353 24.6  1.3 141452 51800 ?        R    18:03   0:06 php bin/console azuracast:sync:run
ubuntu   2366846 23.5  1.3 143632 54488 ?        S    18:19   0:00 php /var/azuracast/www/bin/console azuracast:sync:nowplaying:station dnbradio-test-station`

As a last resort, I changed all stations to "CPU Optimized" but it did not help.

htop:
<img width="1276" alt="image" src="https://github.com/AzuraCast/AzuraCast/assets/3534206/3883fe01-92a4-453f-a50f-63dc627631ff">

**Comment by kotevskimk** [comment]
And the solution?
When to receive/will be update for this?

**Comment by renatopinto89** [comment]
The solution was already posted here
Get a proper VPS/Dedicated Server.
If you're getting the cheap stuff you're gonna get bottom of the barrel performance, it's like the old saying "can't get proper lobster for 5bucks"

Look for dedicated offers on OVH or Hetzner, they have amazing bang for the buck.

**Comment by keybrdist** [comment]
@renatopinto89 The issue for me isn't due to my VPS. I did not have this problem on a smaller AWS instance running a previous version of Azuracast.  Do I need to create a new issue about the odd high cpu usage found on the  php scripts e.g. `azura:sync:nowplaying` commands?

**Comment by simpsss** [comment]
> The solution was already posted here Get a proper VPS/Dedicated Server. If you're getting the cheap stuff you're gonna get bottom of the barrel performance, it's like the old saying "can't get proper lobster for 5bucks"
> 
> Look for dedicated offers on OVH or Hetzner, they have amazing bang for the buck.

Can't you understand that the problem is not the VPS? It's simple: in the previous version, everything worked NORMALLY. The issue with the CPU reaching 100% only occurs AFTER the update. Clearly, the problem is not the VPS.

**Comment by kotevskimk** [comment]
When I roll back with created snapshot to previous version everything is working normal.
Yes I think so, the problem is not the VPS.

**Comment by keybrdist** [comment]
I definitely have a feeling something is wrong somewhere in the PHP command execution causing high CPU usage. 

`php bin/console azuracast:sync:run`  - uses a constant >16% CPU and somehow I have 2 of them running (is this normal)?
```
ubuntu   2413666 16.3  1.3 141452 51804 ?        R    21:03   0:23 php bin/console azuracast:sync:run
ubuntu   2415862 19.0  1.3 141452 51668 ?        R    21:10   0:00 php bin/console azuracast:sync:run
````


`sync:nowplaying` -- uses >20% and I have 4 stations on my system, and while you might think my VPS does not have enough resources to handle this, I have not had this issue until the recent upgrade. Perhaps something else is wrong, but I must assume it's something related to this update since I did not have the issue prior to updating.  P.S. No matter how small the number of tracks the station has the CPU usage is the same. One of these stations has 0 tracks (it's just a relay station) and still has high CPU on a `sync:nowplaying` command.
```
ubuntu   2414438 31.0  1.5 369848 59992 ?        R    21:06   0:00 php /var/azuracast/www/bin/console azuracast:sync:nowplaying:station section8recs
ubuntu   2414439 28.0  1.3 216520 55132 ?        S    21:06   0:00 php /var/azuracast/www/bin/console azuracast:sync:nowplaying:station jungletrain.net
ubuntu   2414440 24.0  1.4 145680 57048 ?        R    21:06   0:00 php /var/azuracast/www/bin/console azuracast:sync:nowplaying:station dnbradio
ubuntu   2414441 27.5  1.5 369196 59740 ?        Rl   21:06   0:00 php /var/azuracast/www/bin/console azuracast:sync:nowplaying:station darkstep
```

It seems no matter which PHP command is being run it uses a minimum 16% CPU until it is finished performing the task.
```
ubuntu   2415872 20.0  1.0 131080 42460 ?        R    21:10   0:00 php /var/azuracast/www/bin/console azuracast:sync:task App\Sync\Task\CheckFolderPlaylistsTask
ubuntu   2415873 20.0  1.0 131080 42328 ?        R    21:10   0:00 php /var/azuracast/www/bin/console azuracast:sync:task App\Sync\Task\CheckPodcastPlaylistsTask
ubuntu   2415874 20.0  1.0 131080 41816 ?        R    21:10   0:00 php /var/azuracast/www/bin/console azuracast:sync:task App\Sync\Task\CheckRequestsTask
ubuntu   2415875 19.0  1.0 131080 42328 ?        R    21:10   0:00 php /var/azuracast/www/bin/console azuracast:sync:task App\Sync\Task\CleanupRelaysTask
ubuntu   2415876 19.0  1.0 131080 41944 ?        R    21:10   0:00 php /var/azuracast/www/bin/console azuracast:sync:task App\Sync\Task\EnforceBroadcastTimesTask
ubuntu   2415877 19.0  1.0 131080 41948 ?        R    21:10   0:00 php /var/azuracast/www/bin/console azuracast:sync:task App\Sync\Task\MoveBroadcastsTask
ubuntu   2415878 18.0  1.0 131080 41816 ?        R    21:10   0:00 php /var/azuracast/www/bin/console azuracast:sync:task App\Sync\Task\QueueInterruptingTracks
ubuntu   2415879 18.0  1.0 131080 41688 ?        R    21:10   0:00 php /var/azuracast/www/bin/console azuracast:sync:task App\Sync\Task\ReactivateStreamerTask
ubuntu   2415880 18.0  1.0 131080 41564 ?        R    21:10   0:00 php /var/azuracast/www/bin/console azuracast:sync:task App\Sync\Task\RunBackupTask
```

Example: RunBackupTask already ran a few minutes ago which means it probably already did the backup routine but each time it runs it takes up ~20% CPU which doesn't make sense, unless there is something fundamentally wrong unrelated to the backup service itself. I suspect the issue is upstream somewhere in the PHP execution.

**Comment by keybrdist** [comment]
@kotevskimk can you try the Rolling Release on a new server just to test it out?  I want to do this, but I don't have time right now.

**Comment by kotevskimk** [comment]
> @kotevskimk can you try the Rolling Release on a new server just to test it out? I want to do this, but I don't have time right now.

I'll try for two days. I am on vacation now.

**Comment by BusterNeece** [comment]
@keybrdist The regular Now Playing operation shouldn't be consuming that much CPU of course, but it's possible that you're seeing higher CPU spikes than before because we've actually fixed bugs related to the parallel operation of the Now Playing sync tasks. Previously, it would run the tasks in sequence, which led to stations getting very out of date on Now Playing data, especially on larger installations; the new method runs the tasks in parallel, which increases overall load but should also improve responsiveness significantly.

There's an environment variable called `NOW_PLAYING_DELAY_TIME` which you can set in `azuracast.env` on the host to increase the amount of time between subsequent checks of Now playing data on each station. This can help reduce some of the CPU load if things are getting checked too often.

For everyone commenting here, please be especially patient with us regarding issues like this, as while we understand things are frustrating for you as users, issues like "CPU is at 100%" are among the most vague and most challenging to diagnose and resolve issues we see. It may not even be possible for us to determine exactly what's overloading things just from our system logs, instead having to rely on profiling to do the job, which is quite a process.

**Comment by Moonbase59** [comment]
Just out of curiosity, I watched my `htop`, filtered by `azuracast:sync`, for a while.
Seems to be rather well-behaving. The CheckMediaTask hits in once in a while (expected), and the nowplaying updates are done fast.

I run 4 stations w/ ~160k tracks and ~70 playlists/station on a Ubuntu 22.04 VM w/ 4 CPUs, 4 GB RAM, 80 GB SSD.

3 minute video watching it live:
https://github.com/AzuraCast/AzuraCast/assets/3706922/8f460d7a-43f6-4456-a5ab-7ae317e93e93

**Comment by keybrdist** [comment]
@Moonbase59 I'm curious what does `sync:nowplaying` actually do under the hood? It sounds like a rather simple operation, but maybe its not.

Also should I have 2 instances of `azuracast:sync:run` running for any reason? 

Seeing every single php operation take up 20% minimum regardless of the complexity of the function makes me think something else is going on affefcting all of them in the same way.

**Comment by BusterNeece** [comment]
@keybrdist It loops through all of your active stations and dispatches a standalone `sync:nowplaying:station StationName` process to update its Now Playing data, keep its queue populated and other maintenance tasks, then supervises all of those subprocesses as they run. Internally, you can configure both how many parallel processes it runs and what the delay is between those processes being spawned after the previous one exits.

**Comment by keybrdist** [comment]
Thanks @BusterNeece for your time -- Where does it get the now playing data from? I would like to find out where in the execution the slowdown might be occurring.

One of the stations doesn't have any playlists, or tracks, no autodj, no shoutcast/icecast server, etc. It's just a single remote relay and only the user facing web-player is used. However, the `sync:nowplaying:station` command still consumes ~20% just like the others.

Another station in our setup is a "test station" with just 1 track added and bare essentials. We use it as a place for our dj's to test their connectivity independently from our main station.  The `sync:nowplaying:station` command on this station also consumes ~20%.

Re: configuring parallel procs.. I assume you mean I can change this in the supervisor config or is it elsewhere?

**Comment by keybrdist** [comment]
I added the following to the .env before restarting:
```
# Custom Config:
NOW_PLAYING_DELAY_TIME=15
NOW_PLAYING_MAX_CONCURRENT_PROCESSES=2
```

I also removed all but 2 stations (completely disabled them) to get the avg under 1.00. 
It stayed below 1.00 for short time but now is back up to >5.00. 
So it seems no matter how many stations I had running it avg around 7.00-10.00 load average.
This is the same avg usage we saw before making these changes.

<img width="708" alt="image" src="https://github.com/AzuraCast/AzuraCast/assets/3534206/520f34a7-9a9e-44c3-9ac9-8a7c3ced3a47">

mariadb cpu usage is high here. I didn't see that before. Is there perhaps a new query introduced recently which needs to be optimized (indexes)?

<img width="1197" alt="image" src="https://github.com/AzuraCast/AzuraCast/assets/3534206/088da537-54db-43ba-b619-c55f071e2885">

<img width="1207" alt="image" src="https://github.com/AzuraCast/AzuraCast/assets/3534206/7cb10a6e-a131-4b4a-be75-8cef118b8b27">

I guess let me know what I can do that will help you.   I would upgrade to a 4 vCPU instance on AWS, but this would cost me ~$170 per month and completely eat up this non-profit's budget. The previous instance I was using was just $30, so you can see why I'm trying to avoid this route.  

Either way, it seems the application code is using more resources than necessary somewhere.. ;)

I will spin up a new instance on Hetzner for testing purposes when I get some more time.

Thanks again for your time.

**Comment by BusterNeece** [comment]
@keybrdist Looks like a fair portion of your CPU load is being stolen (`st`)...sometimes this can indicate that either you're on an oversold VPS or that your VPS has been throttled due to long-term high CPU consumption (which a lot of hosts do to things like AzuraCast/Liquidsoap when you're not on a dedicated-CPU plan)...that may be worth looking into, as that's a very, very frequent cause of sudden CPU spikes like that.

**Comment by keybrdist** [comment]
Yes, I read that in earlier comments. We are on AWS and have burstable CPU limits.  I cannot imagine that we wouldn't be able to run Azuracast on 2 vCPU's like we have been for the last 6 mos.  We even had it running smoothly on 1 vCPU with 4 stations running.  In fact have an instance of OBS running smoothly on a 2 vCPU instance. 

I would not be poking around at the PHP and questioning the optimization there if I didn't see the constant 20% usage on each individual PHP script. For example, the Backup script should not be consuming 20% each time it runs because it shouldn't actually be doing anything except checking if a backup was already run (it has).  So I would like to find out if PHP layer is the actual culprit before completely rethinking my infrastructure.

**Comment by BusterNeece** [comment]
@keybrdist I understand that. Unless you've switched to using one of our development builds, though, our PHP loadout has remained remarkably stable, including with the last version. That 20% CPU figure tells me very very little about what's actually causing it...is there an IO bottleneck causing trouble? Who knows? Only more detailed profiling can tell us that, and right now production images don't ship with profiling as it would slow down performance and increase image size.

I'll continue to investigate locally to see if I find any anomalous results in our profiling, but I haven't seen anything out of the ordinary in everyday operation.

**Comment by keybrdist** [comment]
I deployed a new instance.

Rolling Release, Hetzner, 4 vCPU  

Note: 2 vCPU is usually enough, but was unstable.

`php sync:nowplaying` is spiking still. See the top entry of this screenshot.

### PHP spikes

<img width="531" alt="image" src="https://github.com/user-attachments/assets/a430e873-7f07-49d8-9621-40c6af8a6149">

### Spike `azuracast:sync:nowplaying:station xxxx`

```bash
forge@dnbradio-azura-main:~$ ps -eo pid,pcpu,comm,cmd --sort=-pcpu | grep php | head -n 10
 340948 75.0 php             php /var/azuracast/www/backend/bin/console azuracast:sync:nowplaying:station jungletrain.net
 340941 70.5 php             php /var/azuracast/www/backend/bin/console azuracast:sync:nowplaying:station darkstep
 340939 65.4 php             php /var/azuracast/www/backend/bin/console azuracast:sync:nowplaying:station dnbradio
 340940 55.5 php             php /var/azuracast/www/backend/bin/console azuracast:sync:nowplaying:station section8recs
 340780  0.7 php             php backend/bin/console queue:process --worker-name=app_worker_0
 332522  0.7 php-fpm         php-fpm: pool www
 334895  0.7 php-fpm         php-fpm: pool www
 335401  0.7 php-fpm         php-fpm: pool www
 334492  0.7 php-fpm         php-fpm: pool www
 336163  0.6 php-fpm         php-fpm: pool www
 ```
 
 Note: Some of these stations are bare essentials with nothing turned on.
 
I don't see anything in the logs other than WARNING and ERROR related to HTTP response payloads not having expected properties.

```
[2024-07-11T21:05:31.474039+00:00] AzuraCast.WARNING: Request returned status code 400. {"body":""} {"station":{"id":3,"name":"Section 8 Recordings - 24/7 Dark DnB via @section8recs"}}
[2024-07-11T21:05:31.562111+00:00] AzuraCast.WARNING: Request returned status code 404. {"body":""} {"station":{"id":3,"name":"Section 8 Recordings - 24/7 Dark DnB via @section8recs"}}
[2024-07-11T21:05:31.645866+00:00] AzuraCast.WARNING: Response does not contain a "source" listing; the stream may be hidden or misspelled. {"response":"{\"icestats\":{\"admin\":\"admin@localhost\",\"host\":\"localhost\",\"location\":\"Earth\",\"server_id\":\"Icecast 2.4.4\",\"server_start\":\"Tue, 30 Apr 2024 19:45:17 +0200\",\"server_start_iso8601\":\"2024-04-30T19:45:17+0200\",\"dummy\":null}}"} {"station":{"id":3,"name":"Section 8 Recordings - 24/7 Dark DnB via @section8recs"}}
[2024-07-11T21:05:31.851389+00:00] AzuraCast.ERROR: HTTP client GET call to https://source.dnbradio.com/dnbradio_darkstep.mp3/status-json.xsl produced response NULL [] {"station":{"id":4,"name":"Darkstep DnB Channel - The Darkest Drum & Bass via @dnbradio"}}
[2024-07-11T21:05:35.560232+00:00] AzuraCast.WARNING: Request returned status code 400. {"body":""} {"station":{"id":2,"name":"Plush Recordings - 24/7 Deep, Atmospheric DnB via @plushrecs"}}
[2024-07-11T21:05:35.561337+00:00] AzuraCast.WARNING: Remote server has no mount points listed. {"response":"<?xml version=\"1.0\"?>\n<icestats><admin>webmaster@dnbradio.com</admin><client_connections>5121910</client_connections><clients>0</clients><connections>5132240</connections><file_connections>164</file_connections><host>localhost</host><listener_connections>45882</listener_connections><listeners>0</listeners><location>Earth</location><server_id>Icecast 2.4.4</server_id><server_start>Fri, 09 Feb 2024 04:35:11 +0100</server_start><server_start_iso8601>2024-02-09T04:35:11+0100</server_start_iso8601><source_client_connections>47</source_client_connections><source_relay_connections>0</source_relay_connections><source_total_connections>47</source_total_connections><sources>0</sources><stats>0</stats><stats_connections>0</stats_connections></icestats>\n"} {"station":{"id":2,"name":"Plush Recordings - 24/7 Deep, Atmospheric DnB via @plushrecs"}}
```

Overlap warning:

```
time="2024-07-11T21:03:00Z" level=info msg="job succeeded" iteration=1778 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
time="2024-07-11T21:04:00Z" level=info msg=starting iteration=1779 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
time="2024-07-11T21:05:00Z" level=info msg=starting iteration=1780 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
time="2024-07-11T21:05:00Z" level=warning msg="overlapping jobs: job is still running since 2024-07-11 21:04:00 +0000 UTC (1m0s elapsed)" iteration=1779 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
time="2024-07-11T21:05:01Z" level=info msg="job succeeded" iteration=1780 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
time="2024-07-11T21:06:00Z" level=warning msg="overlapping jobs: job is still running since 2024-07-11 21:04:00 +0000 UTC (2m0s elapsed)" iteration=1779 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
time="2024-07-11T21:06:00Z" level=info msg=starting iteration=1781 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
time="2024-07-11T21:06:03Z" level=info msg="job succeeded" iteration=1781 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
time="2024-07-11T21:07:00Z" level=warning msg="overlapping jobs: job is still running since 2024-07-11 21:04:00 +0000 UTC (3m0s elapsed)" iteration=1779 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
time="2024-07-11T21:07:00Z" level=info msg=starting iteration=1782 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
time="2024-07-11T21:07:01Z" level=info msg="job succeeded" iteration=1782 job.command="azuracast_cli azuracast:sync:run" job.position=0 job.schedule="0 * * * * * *"
```



I know you want some extensive profiling but I'm not able to provide that on this production server. I will setup a new instance soon for testing purposes.
 
 ---
 Side note.  The OP had logs with errors related to `azuracast:sync:nowplaying`. Maybe this is unrelated, but I thought it would be a worthy note.
 
 ```
 [2024-06-17T00:24:13.511699+00:00] AzuraCast.NOTICE: Starting new Message Queue worker process. {"runtime":0,"workerName":"app_worker_0"} []
[2024-06-17T00:24:18.629140+00:00] AzuraCast.ERROR: Message queue error: NOGROUP No such key 'messages.high_priority' or consumer group 'symfony' in XREADGROUP with GROUP option {"workerName":"app_worker_0","exception":"[object] (Symfony\\Component\\Messenger\\Exception\\TransportException(code: 0): NOGROUP No such key 'messages.high_priority' or consumer group 'symfony' in XREADGROUP with GROUP option at /var/azuracast/www/vendor/symfony/redis-messenger/Transport/Connection.php:447)"} []
[2024-06-17T00:24:18.630109+00:00] AzuraCast.WARNING: Console command `azuracast:queue:process` exited with error code 1. [] []
[2024-06-17T00:24:19.934942+00:00] AzuraCast.NOTICE: Failed to define an expiration for the "{resource}" lock, someone else acquired the lock. {"resource":{"Symfony\\Component\\Lock\\Key":"lock_nowplaying_ragostivar"}} []
[2024-06-17T00:24:19.937851+00:00] AzuraCast.ERROR: Symfony\Component\Lock\Exception\LockConflictedException:  (uncaught exception) at /var/azuracast/www/vendor/symfony/lock/Store/RedisStore.php line 166 while running console command `azuracast:sync:nowplaying` {"exception":"[object] (Symfony\\Component\\Lock\\Exception\\LockConflictedException(code: 0):  at /var/azuracast/www/vendor/symfony/lock/Store/RedisStore.php:166)"} []
[2024-06-17T00:24:19.938180+00:00] AzuraCast.WARNING: Console command `azuracast:sync:nowplaying` exited with error code 1. [] []
[2024-06-17T00:24:21.238153+00:00] AzuraCast.NOTICE: Starting new Message Queue worker process. {"runtime":0,"workerName":"app_worker_0"} []
[2024-06-17T00:24:31.370576+00:00] AzuraCast.WARNING: Duplicate prevention yielded no playable song; resetting song queue. [] {"station":{"id":4,"name":"Radio Pela"}}
[2024-06-17T00:24:31.372124+00:00] AzuraCast.WARNING: Playlist "default" did not return a playable track. {"playlist_id":13,"playlist_order":"shuffle","allow_duplicates":false} {"station":{"id":4,"name":"Radio Pela"}}
[2024-06-17T00:24:31.972462+00:00] AzuraCast.ERROR: Request returned status code 400. {"body":""} {"station":{"id":3,"name":"Radio G"}}
```

```
[2024-07-11T20:36:00.529067+00:00] AzuraCast.ERROR: Message queue error: NOGROUP No such key 'messages.media' or consumer group 'symfony' in XREADGROUP with GROUP option {"workerName":"app_worker_0","exception":"[object] (Symfony\\Component\\Messenger\\Exception\\TransportException(code: 0): NOGROUP No such key 'messages.media' or consumer group 'symfony' in XREADGROUP with GROUP option at /var/azuracast/www/vendor/symfony/redis-messenger/Transport/Connection.php:453)"} []
[2024-07-11T20:36:00.529402+00:00] AzuraCast.WARNING: Console command `azuracast:queue:process` exited with error code 1. [] []
[2024-07-11T20:36:01.664916+00:00] AzuraCast.NOTICE: Starting new Message Queue worker process. {"runtime":0,"workerName":"app_worker_0"} []
[2024-07-11T20:37:03.464149+00:00] AzuraCast.ERROR: HTTP client GET call to https://services.getmeradio.com/api/song_update/?token=7018982015255593&station_id=9389&title=Station%20Offline&artist= produced response NULL [] []
[2024-07-11T20:37:03.464640+00:00] AzuraCast.ERROR: /var/azuracast/www/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php L210: cURL error 28: Operation timed out after 3000 milliseconds with 0 bytes received (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://services.getmeradio.com/api/song_update/?token=7018982015255593&station_id=9389&title=Station%20Offline&artist= {"exception":"[object] (GuzzleHttp\\Exception\\ConnectException(code: 0): cURL error 28: Operation timed out after 3000 milliseconds with 0 bytes received (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://services.getmeradio.com/api/song_update/?token=7018982015255593&station_id=9389&title=Station%20Offline&artist= at /var/azuracast/www/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:210)"} []
```

✌️

**Comment by BusterNeece** [comment]
Pushing an update to Rolling Release that should improve things by slowing down concurrency on processes, plus optimizing the looping process manager to use less computationally-intensive functions and smarter defaults.

If you're experiencing CPU overload, it may be helpful to update to this version and let me know if it improves anything.

**Comment by keybrdist** [comment]
Maybe you can close this issue now (not sure).  Things have been stable for over a month now on 4 vCPU Hetzner -- running on Rolling Release #[ba75595](https://github.com/AzuraCast/AzuraCast/commit/ba755951485c9c5ba7322dd1d486db57e8e65049) (2024-07-09 4:02)

**Comment by kotevskimk** [comment]
Is version 20.0.2 correct that problem?

**Comment by chrissarvanis** [comment]
I have the same problem, I have a VPS at Contabo with 8 vCPU Cores and 24 GB RAM.
I have 11 radio stations running with the Rolling Release of AzuraCast.
I had created an old Snapshot that works properly, but after one day, AzuraCast starts overloading at 100%.
The exact same thing happens if I update to the latest version of the Rolling Release.
The problem appeared after version v0.20.1. Before that, everything was working fine without issues.

**Comment by chrissarvanis** [comment]
![cpu](https://github.com/user-attachments/assets/c0cf655f-2de8-4903-916f-8b93e7b93c56)

**Comment by keybrdist** [comment]
> Contabo

I would suggest contacting Contabo as others have done.  For me.. I had to upgrade my VPS to 4vCPU.. It costs more, but that's the only thing that worked.

**Comment by chrissarvanis** [comment]
No it is Azuracast issue, I changed VPS (HostKey.com) and still the issue remains the same.
Also I disabled "Write Playlists to Liquidsoap" and nothing happens!

**Comment by kotevskimk** [comment]
Yes! It is Azuracast issue. I have 24 CPU and still the same.

**Comment by BusterNeece** [comment]
@killemov You can't spin up AzuraCast with **no** configuration at all, as it needs either a root password or instructions to generate a random one (which will be printed in the logs, and which you'll have to save somewhere). See our [environment variables](https://www.azuracast.com/docs/getting-started/settings/#database-configuration) section for details.

**Comment by killemov** [comment]
You CAN, I just did. It will not do anything useful and you can't access anything. BUT ... It will startup mariadbd ... infinitely. Which causes a 100%-ish load.

**Comment by BusterNeece** [comment]
@killemov When I say "you can't" do that, I mean you can't do that without the problem you're experiencing. The container logs should've clearly indicated that environment variables were missing in this scenario.

**Comment by killemov** [comment]
Ah, people have a habit to say you can't (without context, just like you did) when they mean you shouldn't. Portainer makes you look for the logs and it was just an experiment. I believe that the scripts that run within the image should check for the existence and proper values of the environment variables and not start anything not properly configured, including mariadbd.
